src/advisories/Advisories/Import.hs

```diff
@@ -0,0 +1,182 @@
+module Advisories.Import where
+
+import Data.Foldable (forM_, traverse_)
+import Data.Function ((&))
+import Data.List.NonEmpty (NonEmpty)
+import Data.List.NonEmpty qualified as NonEmpty
+import Data.Set qualified as Set
+import Data.UUID.V4 qualified as UUID
+import Data.Vector (Vector)
+import Data.Vector qualified as Vector
+import Effectful
+import Effectful.Error.Static
+import Effectful.PostgreSQL.Transact.Effect (DB)
+import Effectful.Trace
+import Monitor.Tracing qualified as Tracing
+import Security.Advisories.Core.Advisory
+import Security.Advisories.Filesystem (listAdvisories)
+import Validation (Validation (..))
+
+import Advisories.Import.Error
+import Advisories.Model.Advisory.Types
+import Advisories.Model.Advisory.Update qualified as Update
+import Advisories.Model.Affected.Types
+import Advisories.Model.Affected.Update qualified as Update
+import Flora.Import.Package
+import Flora.Model.Package.Guard (guardThatPackageExists)
+import Flora.Model.Package.Types
+import Flora.Model.Release.Guard (guardThatReleaseExists)
+import Flora.Model.Release.Types
+import OSV.Reference.Orphans
+
+processAdvisories
+  :: ( Trace :> es
+     , IOE :> es
+     , DB :> es
+     , Error (NonEmpty AdvisoryImportError) :> es
+     )
+  => FilePath
+  -> Eff es ()
+processAdvisories root = do
+  importResult <- importAdvisories root
+  case importResult of
+    Failure failures ->
+      throwError failures
+    Success advisories ->
+      forM_ advisories $ \advisory -> processAdvisory advisory
```

Comment:
```suggestion
      mapM_ processAdvisory advisories
```

Reply:
@blackheaven I'm interested in how I could become less coupled to `hsec`! :)